js/src/behaviors/tabs.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Tab control for assessment lens
+ * @param container - Container of tab control component
+ */
+/**
+ * Tabs behaviors
+ */
+export function initTabs(container: HTMLElement = document.body) {
+	updateTabNav(container);
+	initTabNavClickListeners(container);
+	initTabControlClickListeners(container);
+}
+
+function initTabNavClickListeners(container: HTMLElement = document.body) {
+	const buttons = Array.from(container.querySelectorAll('.tab-nav')) as HTMLButtonElement[];
```

Comment:
We need tabs to work regardless of when they were added to the DOM. So, I think we'll need to rearrange this function to use window.addEventListener

```ts
		const container=
			event.target instanceof Element && event.target.closest('[data-tabs-control]'); // or whatever the relevant selector is
		if (!container) {
			return;
		}
[... do you stuff]
```
The function will run once no developer should have to worry about re-calling the initTabs function. 

Another viable option would be to create a webcomponent instead.

Reply:
Edit: ignore this in favor of @jdanyow comment about reusing gh's web component

Reply:
Ah this is for tab-nav, which will still require some of our own JS. So we'll like still need this. 

By the way tab-nav isn't currently bubbling the even correctly. Clicking on the span with the arrow does nothing.

Reply:
updated